### PR TITLE
Fixes #6757: Broken link and misplaced page in documentation

### PR DESCRIPTION
**Pull request by fanf**
There is three main action here: 

- remove 80_server_optimization.txt and move the corresponding text in 6_handbook/20_application_tuning.txt
- add two part about jetty: outofmemory errors, so that is someone search for the term in the doc, he finds something, and how to configure the RAM. 
- correct all the links on these parts which were silently broken.

### CODE REVIEW

**Comment by fanf** [comment]
PR updated.

File: 00_introduction/22_architecture_root_server.txt [hunk 1]
```diff
@@ -25,8 +25,8 @@ so for example for 1500-2000 nodes, you need at minimum 6GB of RAM, and
 for 2500-3000 you need 8GB.
 
 The RAM given to the web application must be augmented as explained in 
-the chapter [[jetty, about Jetty configuration]] so that it is
-the half of the RAM of the server, rounded down to the GB.
+the chapter <<_configure_ram_allocated_to_jetty, about webapplication RAM configuration>> 
+so that it is the half of the RAM of the server, rounded down to the GB.
```
Comment (jooooooon) [inline 00_introduction/22_architecture_root_server.txt]: rounded down to the *nearest* GB

File: 6_handbook/20_application_tuning.txt [hunk 2]
```diff
@@ -21,13 +21,137 @@ About the latter, there is some configuration switches that you might want to
...
 
-* -XX:PermSize -XX:MaxPermSize: These parameters are acceptable for most installations,
+* JAVA_MAXPERMSIZE: That parameter is acceptable for most installations,
 but you might want to decrease them a bit if using a machine that is not very powerful /
 RAM abundant. Increasing them is not really useful.
 
+==== Java "Out Of Memory Error"
+
+It may happen that you get java.lang.OutOfMemoryError. 
+There can be of several type,
```
Comment (jooooooon) [inline 6_handbook/20_application_tuning.txt]: s/There/They/

File: 6_handbook/20_application_tuning.txt [hunk 3]
```diff
@@ -21,13 +21,137 @@ About the latter, there is some configuration switches that you might want to
...
+but the most common is: "java.lang.OutOfMemoryError: Java heap space".
+
+This error mean that the web application need more RAM that what was given.
```
Comment (jooooooon) [inline 6_handbook/20_application_tuning.txt]: s/mean/means/
s/need/needs/

Comment (jooooooon) [inline 6_handbook/20_application_tuning.txt]: s/that/than/

File: 6_handbook/20_application_tuning.txt [hunk 4]
```diff
@@ -21,13 +21,137 @@ About the latter, there is some configuration switches that you might want to
...
+It may be linked to a bug where some processing consumed much more memory than
```
Comment (jooooooon) [inline 6_handbook/20_application_tuning.txt]: s/processing/process/

File: 6_handbook/20_application_tuning.txt [hunk 5]
```diff
@@ -21,13 +21,137 @@ About the latter, there is some configuration switches that you might want to
...
+needed, but most of the time, it simply mean that your system grown and need more
```
Comment (jooooooon) [inline 6_handbook/20_application_tuning.txt]: s/mean/means/
s/grown/has grown/
s/need/needs/

File: 6_handbook/20_application_tuning.txt [hunk 6]
```diff
@@ -21,13 +21,137 @@ About the latter, there is some configuration switches that you might want to
...
+memory.
+
+you can follow the configuration step describred in the following paragraphe.
```
Comment (jooooooon) [inline 6_handbook/20_application_tuning.txt]: s/you/You/
s/step/steps/
s/paragraphe/paragraph/

File: 6_handbook/20_application_tuning.txt [hunk 7]
```diff
@@ -21,13 +21,137 @@ About the latter, there is some configuration switches that you might want to
...
+
+==== Configure RAM allocated to Jetty
+
+To change the RAM given to Jetty, you have to: 
+
+----
+
+# edit /opt/rudder/etc/rudder-jetty.conf with you prefered text editor, for example vim:
```
Comment (jooooooon) [inline 6_handbook/20_application_tuning.txt]: s/prefered/preferred/

File: 6_handbook/20_application_tuning.txt [hunk 8]
```diff
@@ -21,13 +21,137 @@ About the latter, there is some configuration switches that you might want to
...
+vim /opt/rudder/etc/rudder-jetty.conf
+
+# modify JAVA_XMX to set the value to your need. 
+# The value is given in MB by default, but you can also use the "G" unit to specify a size in GB.
+
+JAVA_XMX=2G
+
+# save you changes, and restart Jetty:
```
Comment (jooooooon) [inline 6_handbook/20_application_tuning.txt]: s/you/your/

**Comment by jooooooon** [comment]
I have read this PR and approve it.

I have proposed some English language improvements (well, mostly typos) that it would be nice to include. However, if there is a rush to publish this, feel free to merge the PR ASAP and add a later PR with the English fixes.

File: 6_handbook/20_application_tuning.txt (hunk 6 above)
Comment (VinceMacBuche) [inline 6_handbook/20_application_tuning.txt]: s/describred/described/

File: 6_handbook/20_application_tuning.txt [hunk 9]
```diff
@@ -21,13 +21,137 @@ About the latter, there is some configuration switches that you might want to
...
+# The value is given in MB by default, but you can also use the "G" unit to specify a size in GB.
+
+JAVA_XMX=2G
+
+# save you changes, and restart Jetty: 
+service restart rudder-jetty
+
+----
+
+The heuristic to choose the amount of memory needed for the web application is described in <<_hardwareh_specifications, the Rudder hadware
```
Comment (VinceMacBuche) [inline 6_handbook/20_application_tuning.txt]: s/hardwareh/hardware/

Comment (VinceMacBuche) [inline 6_handbook/20_application_tuning.txt]: and 

s/hadware/hardware

**Comment by fanf** [comment]
PR updated for the typos, self merging it.

**Pull request merged**